src/borg/archiver.py

```diff
@@ -433,6 +435,14 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = 0
+            for item in archive.iter_items(filter=lambda item: matcher.match(item.path)):
+                extracted_size += item.file_size()
```

Comment:
I somewhat see the point you're trying to make.

> sparse files

We don't extract those by default (--sparse). If that's specified then I think it can be expected that apparent write speed will vary.

> also, it only considers extracted size - what if the bottleneck is the repo connection and thus the compressed size is what counts?

Hm? It still counts correctly. The speed will vary with repo speed and compression ratio (if it varies a lot), which is kinda clear I think.

> so, I am still rather against progress indication that outputs information that can be very wrong and would prefer outputting only information that is correct for sure.

I don't think this is very wrong, since it says "extracting files" so the percentage relates to how much of the files are extracted. That there are many factors contributing (bandwidths on input/output end, latency on both ends, many small vs few large files, compression ratio if any, varying background/other CPU load etc etc) how fast that is.